### PR TITLE
HOCS-3103 Using new data from somu table rather than case data to sho…

### PR DESCRIPTION
**Pull request by gardnersam**
…w contribution status

### CODE REVIEW

**Review by philbarton — COMMENTED** [review]
This looks ok to me but might be worth asking Liam to have a quick look.

File: src/shared/common/components/workstack.jsx [hunk 1]
```diff
@@ -387,13 +387,13 @@ class WorkstackAllocate extends Component {
             case ColumnRenderer.TRUNCATE_TEXT:
                 return <td key={row.uuid + column.dataValueKey} className='govuk-table__cell govuk-table__cell--truncated' title={value}>{value}</td>;
             case ColumnRenderer.CONTRIBUTIONS_WARNING:
-                if (row.data.CaseContributions) {
-                    const dueContribution = JSON.parse(row.data.CaseContributions)
-                        .filter(contribution => contribution.data && !contribution.data.contributionStatus)
-                        .map(contribution => contribution.data.contributionDueDate)
+                if (row.somu && row.somu.caseContributions) {
+                    const dueContribution = row.somu.caseContributions
+                        .filter(contribution => !JSON.parse(contribution).contributionStatus)
```
Comment (LiamMacP) [inline src/shared/common/components/workstack.jsx:392]: Could you do a map previous to this to do the `JSON.parse` and then filter afterwards? Means you don't repeat the parse?

File: server/lists/adapters/workstacks.js [hunk 2]
```diff
@@ -125,23 +125,24 @@ const highestPriorityContributionStatus = (decoratedContributions) => {
     let highestPriority = 0;
 
     decoratedContributions.map(decoratedContribution => {
```
Comment (LiamMacP) [inline server/lists/adapters/workstacks.js:127]: This should probably be a `forEach`, map creates a new array and is for the transformation of data.

Comment (LiamMacP) [reply server/lists/adapters/workstacks.js:127]: Could also use `reduce` if you are feeling fancy...

File: server/lists/adapters/__tests__/workstacks.spec.js [hunk 3]
```diff
@@ -996,54 +996,61 @@ describe('Workflow Workstack Adapter', () => {
...
-           '{"data":{"contributionDueDate":"2021-06-02","contributionRequestDate":"2021-06-18","contributionStatus":"contributionCancelled"}}]';
         const currentDate = new Date('2021-06-18 12:30');
 
-        const result = decorateContributionsWithStatus(JSON.parse(contributions), currentDate);
-
-        expect(result[0].data['contributionStatus']).toEqual('contributionReceived');
-        expect(result[1].data['contributionStatus']).toEqual('contributionDue');
-        expect(result[2].data['contributionStatus']).toEqual('contributionOverdue');
-        expect(result[3].data['contributionStatus']).toEqual('contributionCancelled');
+        var contributionOne = '{"contributionRequestDate":"2021-06-18","contributionDueDate":"2021-06-02","contributionStatus":"contributionReceived"}';
```
Comment (zuzak) [inline server/lists/adapters/__tests__/workstacks.spec.js:1001]: Use `let` or `const`, not `var`.

Comment (gardnersam) [reply server/lists/adapters/__tests__/workstacks.spec.js]: Updated

File: server/lists/adapters/workstacks.js (hunk 2 above)
Comment (gardnersam) [reply server/lists/adapters/workstacks.js]: Agreed, gone for forEach, apologies for reduced fanciness levels

File: src/shared/common/components/workstack.jsx (hunk 1 above)
Comment (gardnersam) [reply src/shared/common/components/workstack.jsx]: Yep think it increases readability

**Review by LiamMacP — APPROVED** [review]

**Pull request merged**